core/che-core-api-core/src/main/java/org/eclipse/che/api/core/util/HttpJsonRequestUtil.java

```diff
@@ -0,0 +1,123 @@
+/*
+ * Copyright (c) 2012-2018 Red Hat, Inc.
+ * All rights reserved. This program and the accompanying materials
+ * are made available under the terms of the Eclipse Public License v1.0
+ * which accompanies this distribution, and is available at
+ * http://www.eclipse.org/legal/epl-v10.html
+ *
+ * Contributors:
+ *   Red Hat, Inc. - initial API and implementation
+ */
+package org.eclipse.che.api.core.util;
+
+import static java.lang.Integer.parseInt;
+import static java.lang.String.format;
+import static java.util.regex.Pattern.compile;
+
+import com.google.gson.JsonElement;
+import com.google.gson.JsonObject;
+import com.google.gson.JsonParser;
+import com.google.gson.JsonSyntaxException;
+import java.util.regex.Matcher;
+import java.util.regex.Pattern;
+import org.eclipse.che.commons.annotation.Nullable;
+
+/**
+ * Provides useful methods to simplify work with {@link
+ * org.eclipse.che.api.core.rest.HttpJsonRequest} instances.
+ *
+ * @author Roman Nikitenko
```

Comment:
Look like specific to KeycloakOAuthAuthenticationService.java class. Plese move it close to service

Reply:
In DefaultHttpJsonRequest class we create formatted error message with corresponding info when request is failed.
We should have ability to extract this info from formatted message. 
At the moment I use this class to extract and analyze code and error message of response in KeycloakOAuthAuthenticationService.java class only.
But we can face the need to extract formatted in DefaultHttpJsonRequest class info in another services.
So I think this util class more related to DefaultHttpJsonRequest class, but up to you.
WDYT?